dbt_platform_helper/domain/maintenance_page.py

```diff
@@ -84,164 +69,19 @@ def get_env_ips(vpc: str, application_environment: Environment) -> List[str]:
-                    "Type": "fixed-response",
-                    "FixedResponseConfig": {
-                        "StatusCode": "503",
-                        "ContentType": "text/html",
-                        "MessageBody": maintenance_page_content,
-                    },
-                }
-            ],
-            Tags=[
-                {"Key": "name", "Value": "MaintenancePage"},
-                {"Key": "type", "Value": template},
-            ],
-        )
-    except Exception as e:
-        deleted_rules = clean_up_maintenance_page_rules(session, listener_arn)
-        raise FailedToActivateMaintenancePageException(
-            app, env, f"{e}:\n {traceback.format_exc()}", deleted_rules
-        )
-
-
-def clean_up_maintenance_page_rules(
-    session: boto3.Session, listener_arn: str, fail_when_not_deleted: bool = False
-) -> dict[str, bool]:
-    lb_client = session.client("elbv2")
-
-    rules = lb_client.describe_rules(ListenerArn=listener_arn)["Rules"]
-    tag_descriptions = get_rules_tag_descriptions(rules, lb_client)
-
-    deletes = {}
-    for name in ["MaintenancePage", "AllowedIps", "BypassIpFilter", "AllowedSourceIps"]:
-        deleted = delete_listener_rule(tag_descriptions, name, lb_client)
-        deletes[name] = bool(deleted)
-        if fail_when_not_deleted and name == "MaintenancePage" and not deleted:
-            raise ListenerRuleNotFoundException()
-
-    return deletes
-
-
-def remove_maintenance_page(session: boto3.Session, listener_arn: str):
-    clean_up_maintenance_page_rules(session, listener_arn, True)
-
-
 class MaintenancePage:
     def __init__(
         self,
         application: Application,
         io: ClickIOProvider = ClickIOProvider(),
-        get_https_listener_for_application: Callable[
-            [boto3.Session, str, str], str
-        ] = get_https_listener_for_application,
-        # TODO refactor get_maintenance_page_type, add_maintenance_page, remove_maintenance_page into MaintenancePage class with LoadBalancerProvider as the dependency
-        get_maintenance_page_type: Callable[
-            [boto3.Session, str], Union[str, None]
-        ] = get_maintenance_page_type,
+        load_balancer_provider: LoadBalancerProvider = LoadBalancerProvider,
         get_env_ips: Callable[[str, Environment], List[str]] = get_env_ips,
-        add_maintenance_page: Callable[
-            [boto3.Session, str, str, str, List[Service], tuple, str], None
-        ] = add_maintenance_page,
-        remove_maintenance_page: Callable[[boto3.Session, str], None] = remove_maintenance_page,
     ):
         self.application = application
-        self.get_https_listener_for_application = get_https_listener_for_application
         self.io = io
-        self.get_maintenance_page_type = get_maintenance_page_type
+        self.load_balancer_provider = load_balancer_provider  # TODO requires session from environment in application object which is only known during method execution
```

Comment:
we have an inconsistent approach to how we initialise providers, this is partly due to how we manage the Application object and should be revisted.